### PR TITLE
[feat] Support dictionaries as argument of `Run.track`

### PR DESCRIPTION
**Pull request by alberttorosyan**
- Changed the value validation logic and `_track` implementation.
- Added unit-tests to cover dict tracking + negative cases with specifying `name` argument.
- Changed `examples/pytorch_track.py` to use the new interface.

### CODE REVIEW

**Comment by alberttorosyan** [comment]
resolves #2087

File: aim/sdk/tracker.py [hunk 1]
```diff
@@ -116,23 +116,18 @@ def _track(
...
             ctx = Context(context)
-            seq_info = self.sequence_infos[ctx.idx, name]
-            if not seq_info.initialized:
-                self._init_sequence_info(ctx.idx, name, val)
-            step = step if step is not None else seq_info.count
+            for name, val in values.items():
+                seq_info = self.sequence_infos[ctx.idx, name]
+                if not seq_info.initialized:
+                    self._init_sequence_info(ctx.idx, name, val)
+                step = step if step is not None else seq_info.count
```
Comment (mihran113) [inline aim/sdk/tracker.py:126]: doesn't this logic need to be changed, if dictionary is provided?

Comment (alberttorosyan) [reply aim/sdk/tracker.py:126]: nope. all the operations, including getting sequence info, updating, etc. are done on the level of dict item, which is the same as it was before.
so if you track a dict of 5 items, 5 sequence infos should be checked.

**Review by mihran113 — APPROVED** [review]
lgtm

**Pull request merged**